ext/libclementine-tagreader/tagreader.cpp

```diff
@@ -130,6 +132,57 @@ const char* kASF_OriginalDate_ID = "WM/OriginalReleaseTime";
 const char* kASF_OriginalYear_ID = "WM/OriginalReleaseYear";
 }  // namespace
 
+// Helpers for GuessArtistAndTitle()
+static std::string withoutExtension(const std::string  s) {
+  if (s.empty()) return s;
+  const int i = s.find_last_of('.');
+  if (i == std::string::npos) return s;
+  return s.substr(0, i);
+}
+
+static void changeUnderscores(std::string& s) {
+  std::replace(s.begin(), s.end(), '_', ' ');
+}
+
+static std::string& ltrim(std::string& str, const std::string& chars = "\t\n\v\f\r ") {
```

Comment:
Given all the reimplementations you're doing here, you may as well convert to `QString` and use all the helpers there and then convert back to a `std::string` at the end.

Reply:
ok, will do